pkg/validation/validate.go

```diff
@@ -112,6 +112,20 @@ func ValidateLabels(limits LabelValidationLimits, tenantID string, ls []*typesv1
 	if len(ls) == 0 {
 		return NewErrorf(MissingLabels, MissingLabelsErrorMsg)
 	}
+
+	sanitizedNames := make(map[string]bool)
+	//Check if we only have service.name label before sanitization
+	serviceNameValue := phlaremodel.Labels(ls).Get(phlaremodel.LabelNameServiceName)
```

Comment:
While looking for when the labels are sorted, I noticed that we already add unspecified as a value, so this if can never be triggered:

https://github.com/grafana/pyroscope/blob/efeabde8456ee1e1d68db0ceb026305f753866b7/pkg/distributor/distributor.go#L293-L295

Reply:
Not really. It's a safe check... in case anytime this is changed or we ingest labels throughout another path. However, this is likely not going to happen. I'll remove it.

Reply:
I don't think this has any benefit how it is implemented right now: The distributor will always go through. `PushParsed`(which is the only caller of `ValidateLabels`).

I think we need to test it form that level, it is not super straight forward, but that's how I tested your change:

https://github.com/grafana/pyroscope/commit/2535c8d1afc6e0d384be454fcc2a3651d07789aa

You will find that without the returning of the labels it will panic. This is because you modify the label slice without returning the new slice.